Website/src/rendering/src/components/Checkout/PanelBillingAddress.tsx

```diff
@@ -1,30 +1,66 @@
 import useOcCurrentCart from '../../hooks/useOcCurrentCart';
 import { DBuyerAddress } from '../../models/ordercloud/DBuyerAddress';
 import { removeBillingAddress, saveBillingAddress } from '../../redux/ocCurrentCart';
 import { useAppDispatch } from '../../redux/store';
-import { useState } from 'react';
-import AddressCard from './AddressCard';
+import { useEffect, useState } from 'react';
 import { isSameAddress } from '../../helpers/AddressHelper';
+import useOcAddressBook from '../../hooks/useOcAddressBook';
+import useOcAuth from '../../hooks/useOcAuth';
+import CheckoutAddressList from './CheckoutAddressList';
 
 const PanelBillingAddress = (): JSX.Element => {
-  // TODO: this component should also allow choosing a saved address
   const dispatch = useAppDispatch();
+  const [loading, setLoading] = useState(false);
+  const { isAnonymous } = useOcAuth();
   const { order, shippingAddress } = useOcCurrentCart();
+
   const billingAddress = order?.BillingAddress;
-  const [isEditing, setIsEditing] = useState(false);
-  const [loading, setLoading] = useState(false);
+
   const [isSameAsBilling, setIsSameAsBilling] = useState(
     Boolean(billingAddress && shippingAddress && isSameAddress(billingAddress, shippingAddress))
   );
+  const [activeAddressId, setActiveAddressId] = useState(
+    billingAddress?.ID && !isSameAsBilling ? billingAddress.ID : ''
+  );
   const [tempAddress, setTempAddress] = useState({} as DBuyerAddress); // saves address prior to unchecking "same as billing" so we can revert to last saved address if necessary
+  const { saveAddress, addresses } = useOcAddressBook({
+    pageSize: 10,
+    filters: { Editable: true }, // personal addresses
+  });
 
-  const handleSetBillingAddress = async (address: Partial<DBuyerAddress>) => {
+  const isShipOrder = order?.xp?.DeliveryType === 'Ship';
+
+  useEffect(() => {
+    setIsSameAsBilling(
+      Boolean(billingAddress && shippingAddress && isSameAddress(billingAddress, shippingAddress))
+    );
+  }, [billingAddress, shippingAddress]);
+
+  useEffect(() => {
+    setActiveAddressId(billingAddress?.ID && !isSameAsBilling ? billingAddress.ID : '');
+  }, [billingAddress, isSameAsBilling]);
+
+  let allAddresses = [...addresses];
+  if (billingAddress && !billingAddress.ID) {
+    // include one time address
+    allAddresses = [...allAddresses, billingAddress];
+  }
+
+  const handleSetBillingAddress = async (address: DBuyerAddress) => {
     setLoading(true);
+    setActiveAddressId(address?.ID);
     await dispatch(saveBillingAddress(address));
     setLoading(false);
   };
 
+  const handleSaveAddress = async (address: DBuyerAddress) => {
+    address.Shipping = true;
+    address.Billing = true;
+    setLoading(true);
+    const updatedAddress = await saveAddress(address);
+    await handleSetBillingAddress(updatedAddress);
+  };
+
   const handleSameAsShipping = async () => {
     const shouldUseSameAsBilling = !isSameAsBilling;
```

Comment:
The code in `handleSameAsShipping` seems to allow to select both this option and another saved or new address. Maybe we should not toggle "Same as shipping" and always set it as selected when clicked.

Reply:
Actual: Clicking multiple times on "Same as shipping" toggles the active state of this card.
Expected: All clicks on "Same as shipping" select this card as active and unselect the previously selected card.

Actual: When "Same as shipping" card is active, the user can set a second active card among the saved addresses or a new address.
Expected: When clicking on another address card, "Same as shipping" is unselected.